pkg/runner/local_docker.go

```diff
@@ -83,371 +84,80 @@ type LocalDockerRunner struct {
+	localCommonHealthcheck(ctx, hh, cli, ow, r.controlNetworkID, engine.EnvConfig().SrcDir, r.outputsDir)
 
-	ci, err = docker.CheckContainer(ctx, log, cli, "testground-sidecar")
-	if err == nil {
-		switch {
-		case ci == nil:
-			msg := "sidecar container: non-existent"
-			sidecarContainerCheck = api.HealthcheckItem{Name: "sidecar-container", Status: api.HealthcheckStatusFailed, Message: msg}
-		case ci.State.Running:
-			msg := "sidecar container: running"
-			sidecarContainerCheck = api.HealthcheckItem{Name: "sidecar-container", Status: api.HealthcheckStatusOK, Message: msg}
-		default:
-			msg := fmt.Sprintf("sidecar container: status %s", ci.State.Status)
-			sidecarContainerCheck = api.HealthcheckItem{Name: "sidecar-container", Status: api.HealthcheckStatusFailed, Message: msg}
-		}
+	dockerSock := "/var/run/docker.sock"
+	if host := cli.DaemonHost(); strings.HasPrefix(host, "unix://") {
+		dockerSock = host[len("unix://"):]
 	} else {
-		msg := fmt.Sprintf("sidecar container errored: %s", err)
-		sidecarContainerCheck = api.HealthcheckItem{Name: "sidecar-container", Status: api.HealthcheckStatusAborted, Message: msg}
+		ow.Warnf("guessing docker socket as %s", dockerSock)
 	}
 
-	report := &api.HealthcheckReport{
-		Checks: []api.HealthcheckItem{
-			ctrlNetCheck,
-			grafanaContainerCheck,
-			outputsDirCheck,
-			prometheusContainerCheck,
-			pushgatewayContainerCheck,
-			redisContainerCheck,
-			redisExporterContainerCheck,
-			sidecarContainerCheck,
+	sidecarContainerOpts := docker.EnsureContainerOpts{
+		ContainerName: "testground-sidecar",
+		ContainerConfig: &container.Config{
+			Image:      "ipfs/testground:latest",
+			Entrypoint: []string{"testground"},
+			Cmd:        []string{"sidecar", "--runner", "docker", "--pprof"},
+			Env:        []string{"REDIS_HOST=testground-redis", "GODEBUG=gctrace=1"},
+		},
+		HostConfig: &container.HostConfig{
+			PublishAllPorts: true,
+			// Port binding for pprof.
+			PortBindings: nat.PortMap{"6060": []nat.PortBinding{{HostIP: "0.0.0.0", HostPort: "0"}}},
+			NetworkMode:  container.NetworkMode(r.controlNetworkID),
+			// To lookup namespaces. Can't use SandboxKey for some reason.
+			PidMode: "host",
+			// We need _both_ to actually get a network namespace handle.
+			// We may be able to drop sys_admin if we drop netlink
+			// sockets that we're not using.
+			CapAdd: []string{"NET_ADMIN", "SYS_ADMIN"},
+			// needed to talk to docker.
+			Mounts: []mount.Mount{{
+				Type:   mount.TypeBind,
+				Source: dockerSock,
+				Target: "/var/run/docker.sock",
+			}},
+			Resources: container.Resources{
+				Ulimits: []*units.Ulimit{
+					{Name: "nofile", Hard: InfraMaxFilesUlimit, Soft: InfraMaxFilesUlimit},
+				},
+			},
+			RestartPolicy: container.RestartPolicy{
+				Name: "unless-stopped",
+			},
+		},
+		ImageStrategy: docker.ImageStrategyBuild,
+		BuildImageOpts: &docker.BuildImageOpts{
+			Name:     "testground-sidecar:latest",
```

Comment:
small bug, the image built is called testground-sidecar:latest, but the image the container looks for is called ipfs/testground. Changed to ipfs/testground for both.